ssl/ssl_session.cc

```diff
@@ -879,6 +892,7 @@ static bool add_session_locked(SSL_CTX *ctx, UniquePtr<SSL_SESSION> session) {
                           /*lock=*/false)) {
         break;
       }
+      ssl_update_counter(ctx, ctx->stats.sess_cache_full, /*lock=*/ false);
```

Comment:
this is the only place where we provide `false` for the lock argument, why is that?

Reply:
This placement is within the function `add_session_locked`. `add_session_locked` in this context means that the mutex has already been locked before this function is called. 
I added a "lock" argument to `ssl_update_counter` to indicate whether the mutex has been locked already, to prevent the mutex from being locked twice. There's also a similar function above called `remove_session` which applies the same concept.

Reply:
can we update the variable name (`locked` to `lock`) and function description then to make this more clear, right now it says:
```
// ssl_update_counter updates the stat counters in |SSL_CTX|. `locked` should be
// set to false if the mutex has already been locked.
void ssl_update_counter(SSL_CTX *ctx, int &counter, bool locked);
```

Reply:
Updated :)